go/vt/tlstest/tlstest.go

```diff
@@ -19,207 +19,360 @@ limitations under the License.
 package tlstest
 
 import (
+	"bytes"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
 	"fmt"
+	"math/big"
+	"net"
 	"os"
-	"os/exec"
 	"path"
+	"strconv"
+	"time"
 
 	"vitess.io/vitess/go/vt/log"
 )
 
 const (
 	// CA is the name of the CA toplevel cert.
-	CA = "ca"
-
-	caConfigTemplate = `
-[ ca ]
-default_ca = default_ca
-
-[ default_ca ]
-database = %s
-default_md = sha256
-default_crl_days = 30
-
-[ req ]
- default_bits           = 4096
- default_keyfile        = keyfile.pem
- distinguished_name     = req_distinguished_name
- attributes             = req_attributes
- x509_extensions       = req_ext
- prompt                 = no
- output_password        = mypass
-[ req_distinguished_name ]
- C                      = US
- ST                     = California
- L                      = Mountain View
- O                      = Vitessio
- OU                     = Vitess
- CN                     = CA
- emailAddress           = test@email.address
-[ req_attributes ]
- challengePassword      = A challenge password
-[ req_ext ]
- basicConstraints       = CA:TRUE
-
-`
-
-	certConfig = `
-[ req ]
- default_bits           = 4096
- default_keyfile        = keyfile.pem
- distinguished_name     = req_distinguished_name
- attributes             = req_attributes
- req_extensions        = req_ext
- prompt                 = no
- output_password        = mypass
-[ req_distinguished_name ]
- C                      = US
- ST                     = California
- L                      = Mountain View
- O                      = Vitessio
- OU                     = Vitess
- CN                     = %s
- emailAddress           = test@email.address
-[ req_attributes ]
- challengePassword      = A challenge password
-[ req_ext ]
- basicConstraints       = CA:TRUE
- subjectAltName         = @alternate_names
- extendedKeyUsage       =serverAuth,clientAuth
-[ alternate_names ]
- DNS.1                  = localhost
- DNS.2                  = 127.0.0.1
- DNS.3                  = %s
-`
+	CA          = "ca"
+	permissions = 0700
```

Comment:
Nit: Go also supports `0o700` now for octal literals now which I usually encourage for clarity. But no big deal either way.